scheduling/round_robin_scheduling_algorithm.py

```diff
@@ -0,0 +1,102 @@
+"""
+Round Robin is a CPU scheduling algorithm.
+In Round Robin each process is assigned a fixed time slot in a cyclic way.
+https://en.wikipedia.org/wiki/Round-robin_scheduling
+"""
+
+
+def calculate_waiting_time(processes, n, burst_time, waiting_time, quantum):
+    """
+    This function calculates the waiting time of some processes that have a
+    specified duration time.
+        Return: The waiting time for each process.
+    >>> calculate_waiting_times([5, 10, 15])
+    [0, 5, 15]
+    >>> calculate_waiting_times([1, 2, 3, 4, 5])
+    [0, 1, 3, 6, 10]
+    >>> calculate_waiting_times([10, 3])
+    [0, 10]
+    """
+    rem_burst_time = [0] * n
+    for i in range(n):
+        rem_burst_time[i] = burst_time[i]
+    t = 0
+    while 1:
+        done = True
+        for i in range(n):
+            if rem_burst_time[i] > 0:
+                done = False
+                if rem_burst_time[i] > quantum:
+                    t += quantum
+                    rem_burst_time[i] -= quantum
+                else:
+                    t = t + rem_burst_time[i]
+                    waiting_time[i] = t - burst_time[i]
+                    rem_burst_time[i] = 0
+        if done is True:
+            break
+
+
+def calculate_turn_around_time(
+    processes, n, burst_time, waiting_time, turn_around_time
+):
+
+    """
+    This function calculates the turnaround time of some processes.
+        Return: The time difference between the completion time and the
```

Comment:
This function should end with a Python `return` statement that provides the caller the turnaround times...

`turnaround_times = calculate_turnaround_times()`